pkg/kubectl/cmd/util/factory.go

```diff
@@ -355,9 +405,9 @@ func (f *factory) DiscoveryClient() (discovery.CachedDiscoveryInterface, error)
 	return NewCachedDiscoveryClient(discoveryClient, cacheDir, time.Duration(10*time.Minute)), nil
 }
 
-func (f *factory) Object() (meta.RESTMapper, runtime.ObjectTyper) {
+func Object(dcf DiscoveryClientFactory, ccf ClientCacheFactory) (meta.RESTMapper, runtime.ObjectTyper) {
```

Comment:
No, we broke the requirements that Factory had on it with the refactor.  Factory was a base class with a subclass possible.  The refactor broke the subclass.  Restoring the subclass behavior is correcting the bug.